Userland/Libraries/LibPDF/Function.cpp

```diff
@@ -18,11 +18,121 @@ struct Bound {
     float upper;
 };
 
+// 3.9.1 Type 0 (Sampled) Functions
 class SampledFunction final : public Function {
 public:
+    static PDFErrorOr<NonnullRefPtr<SampledFunction>> create(Document*, Vector<Bound> domain, Optional<Vector<Bound>> range, NonnullRefPtr<StreamObject>);
     virtual PDFErrorOr<ReadonlySpan<float>> evaluate(ReadonlySpan<float>) const override;
+
+private:
+    Vector<Bound> m_domain;
+    Vector<Bound> m_range;
+
+    Vector<unsigned> m_size;
+    int m_bits_per_sample;
+
+    enum class Order {
+        Linear = 1,
+        Cubic = 3,
+    };
+    Order m_order;
```

Comment:
Can these two get initializers here?

Reply:
Sure, done, but they're always overwritten in the factory method anyways.